src/sidebar/components/group-list-item.js

```diff
@@ -22,53 +39,82 @@ function GroupListItem({ analytics, group }) {
+    const message = `Are you sure you want to leave the group "${group.name}"?`;
+    if (window.confirm(message)) {
+      analytics.track(analytics.events.GROUP_LEAVE);
+      groups.leave(group.id);
+    }
+  };
+
+  const toggleSubmenu = event => {
+    event.stopPropagation();
+
+    // Prevents group items opening a new window when clicked.
+    // TODO - Fix this more cleanly in `MenuItem`.
+    event.preventDefault();
+
+    setExpanded(!isExpanded);
+  };
+
+  // Close the submenu when any clicks happen which close the top-level menu.
+  const collapseSubmenu = () => setExpanded(false);
 
   return (
-    <div
-      className={classnames({
-        'group-list-item__item': true,
-        'is-selected': isSelected,
-      })}
-      onClick={focusGroup}
-      tabIndex="0"
-    >
-      {/* the group icon */}
-      <div className="group-list-item__icon-container">
-        {group.logo && (
-          <img
-            className="group-list-item__icon group-list-item__icon--organization"
-            alt={groupOrgName}
-            src={group.logo}
-          />
-        )}
-      </div>
-      {/* the group name */}
-      <div className="group-list-item__details">
-        <a
-          className="group-list-item__name-link"
-          href=""
-          title={
-            group.type === 'private'
-              ? `Show and create annotations in ${group.name}`
-              : 'Show public annotations'
-          }
-        >
-          {group.name}
-        </a>
-      </div>
-    </div>
+    <Fragment>
+      <MenuItem
+        icon={group.logo || null}
+        iconAlt={orgName(group)}
+        isDisabled={!isSelectable}
+        isExpanded={isExpanded}
+        isSelected={isSelected}
+        isSubmenuVisible={isExpanded}
+        label={group.name}
+        onClick={isSelectable ? focusGroup : toggleSubmenu}
+        onToggleSubmenu={toggleSubmenu}
+      />
+      {isExpanded && (
+        <Fragment>
+          <ul onClick={collapseSubmenu}>
+            {activityUrl && (
+              <li>
+                <MenuItem
+                  href={activityUrl}
+                  icon="share"
+                  isSubmenuItem={true}
+                  label="View group activity"
+                />
+              </li>
+            )}
+            {canLeaveGroup && (
+              <li>
+                <MenuItem
+                  icon="leave"
+                  isSubmenuItem={true}
+                  label="Leave group"
+                  onClick={leaveGroup}
+                />
+              </li>
+            )}
+          </ul>
+          {!isSelectable && (
+            <p className="group-list-item__footer">
+              This group is restricted to specific URLs.
+            </p>
+          )}
+        </Fragment>
+      )}
+    </Fragment>
   );
 }
 
 GroupListItem.propTypes = {
   group: propTypes.object.isRequired,
 
   analytics: propTypes.object.isRequired,
+  groups: propTypes.object.isRequired,
```

Comment:
What is `groups`? Aha! I got confused between a list of groups and the groups service. This is the latter.

Reply:
Yeah. It is a bit confusing in this context ... but we can't rename it here because dependency injection looks up the service by name. It can be aliased to another name in the destructuring that happens in the `function GroupListItem(...)` line though, eg. `{ groups: groupsService, ... }`